### PR TITLE
[KOGITO-5680] Quarkus update Jenkins jobs

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5680

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/233
- https://github.com/kiegroup/kogito-runtimes/pull/1514
- https://github.com/kiegroup/optaplanner/pull/1503
- https://github.com/kiegroup/kogito-examples/pull/908
- https://github.com/kiegroup/optaplanner-quickstarts/pull/188

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

**Comment by radtriste** [comment]
Example quarkus update PR: https://github.com/radtriste/optaplanner/pull/38

**Review by Ginxo — APPROVED** [review]
LGTM

File: .ci/jenkins/dsl/jobs.groovy [hunk 1]
```diff
@@ -87,6 +88,12 @@ if (Utils.isLTSBranch(this)) {
     setupNativeLTSJob()
 }
 
+// Tools folder
+KogitoJobUtils.createQuarkusUpdateToolsJob(this, 'optaplanner', 'Optaplanner', [
```
Comment (rsynek) [inline .ci/jenkins/dsl/jobs.groovy:92]:
```suggestion
KogitoJobUtils.createQuarkusUpdateToolsJob(this, 'optaplanner', 'OptaPlanner', [
```

**Review by rsynek — APPROVED** [review]

**Comment by radtriste** [comment]
Example PR: https://github.com/radtriste/optaplanner/pull/39

**Comment by radtriste** [comment]
`Jenkins Tests / dsl-tests` will be ok once https://github.com/kiegroup/kogito-pipelines/pull/233 is merged

**Pull request merged**